CONTRIBUTING.md

```diff
@@ -357,6 +357,14 @@ See the [Fixing an Issue](#fixing-an-issue) section above for further details.
 Be sure to include unit and integration tests for your new resource under the Tests folder as well as an example under the examples folder.
 Please also add a full description of your new resource and its parameters to the module's README.
 
+### Resource Naming
+All mof-based resource (with Get/Set/Test-TargetResource and a schema.mof file) files should have MSFT_ appended before the resource name (e.g. MSFT_xResource.psm1). This is per a convention that the name (or abbreviated name) of the company that made the resource should be included in the name of mof-based resource files. The friendly name of the resource, defined in the mof file, should not have the MSFT_ prefix.
+
+Composite resource (with a configuration and a .psd1 file) files must have the exact same name as the resource or they will not be able to import. Hence, composite resource files should not have the MSFT_ prefix (e.g. xResource.psm1).
+
+If you are adding a new resource to an experimental/preview module (module name starts with 'x'), the resource name must also start with 'x' (e.g. MSFT_xResource.psm1 or xResource.psm1).
+If you are adding a new resource to a high quality module (module name does not start with 'x' and ends with 'Dsc'), the resource name should not start with 'x' (e.g. MSFT_Resource.psm1 or Resource.psm1). 
```

Comment:
All these explanations in parentheses made me think whether we should have some kind of glossary where we would explain most common terms/abbreviations we use in DscResources area (preview module/high quality module/mof-based resource/composite resource/class resource etc.), so that people new to it could quickly get familiar with those and better understand what we are talking about in issues/PRs?
Something to think about...